docs/how-to-guides/upgrade-bootstrap-kubelet.md

```diff
@@ -0,0 +1,113 @@
+  - [Step 3: Upgrade Kubelet on node](#step-3-upgrade-kubelet-on-node)
+  - [Step 4: Verify](#step-4-verify)
+- [Caveats](#caveats)
+
+## Introduction
+
+[Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) is a daemon
+that runs on every node and it is responsible for managing Pods on the node.
+
+Lokomotive cluster runs two different sets of Kubelet processes. Initially, **bootstrap** Kubelet
+configured on the node as a `rkt` pod joins the cluster, and then Kubelet pod managed using
+DaemonSet (self-hosted Kubelet) takes over the bootstrap Kubelet. Self-hosted Kubelet allows
+seamless updates between Kubernetes patch version for nodes and also to tune nodes configuration
+using tools like `kubectl`.
+
+At the time of writing `lokoctl` cannot update bootstrap Kubelet, so this document explains how to
+perform this operation manually.
+
+## Steps
+
+Perform the following steps on each node, one node at a time.
+
+### Step 1: Drain the node
+
+> **Caution:** If you are using a local directory as a storage for a workload, it will be disturbed
+> by this operation. To avoid this move the workload to another node and let the application
+> replicate the data. If the application does not support data replication across instances, then
+> expect downtime.
+
+```bash
+kubectl drain --ignore-daemonsets <node name>
+```
+
+### Step 2: Find out the IP and SSH
+
+Find the IP of the node by visiting the cloud provider dashboard. Then, connect to selected machine
+using SSH.
+
+```bash
+ssh core@<IP Address>
+```
+
+### Step 3: Upgrade Kubelet on node
+
+Run the following commands:
+
+> **NOTE**: Before proceeding to other commands, set the `latest_kube` variable to the latest
+> Kubernetes version. Latest Kubernetes version can be found by running this command after a cluster
+> upgrade: `kubectl version -ojson | jq -r '.serverVersion.gitVersion'`.
+
+```bash
+export latest_kube=<latest kubernetes version e.g. v1.18.0>
+sudo sed -i "s|$(grep -i kubelet_image_tag /etc/kubernetes/kubelet.env)|KUBELET_IMAGE_TAG=${latest_kube}|g" /etc/kubernetes/kubelet.env
+sudo systemctl restart kubelet
+sudo journalctl -fu kubelet
+```
+
+Check the logs carefully. If Kubelet fails to restart and instructs to do something (e.g. deleting
+some file), follow the instructions and reboot the node:
+
+```bash
+sudo reboot
+```
+
+### Step 4: Verify
+
+**When `disable_self_hosted_kubelet` is `true`**:
+
+Once the node reboots and Kubelet rejoins the cluster, output of following command will show new
+version across the node name:
+
+```bash
+kubectl get nodes
+```
+
+**When `disable_self_hosted_kubelet` is `false`**:
+
+Verify that the Kubelet service is in active (running) state:
+
+```bash
+sudo systemctl status --no-pager kubelet
+```
+
+Run the following command to see logs of the process since the last restart:
+
+```bash
+sudo journalctl _SYSTEMD_INVOCATION_ID=$(sudo systemctl \
```

Comment:
What does `_SYSTEMD_INVOCATION_ID` do? Why is it needed here?

Reply:
> **What does `_SYSTEMD_INVOCATION_ID` do?**

From the docs:
> The invocation ID for the runtime cycle of the unit the message was generated in, as available to processes of the unit in $INVOCATION_ID

So everytime a service is started it has an invocation ID.

> **Why is it needed here?**

It is needed here because we want to see logs since last restart and not the logs since the machine was created.

Reply:
Seems a bit overly complex to me, but make sense overall, so all good. Thanks for explanation!